packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1584,9 +1584,20 @@ abstract class ParentDataWidget<T extends RenderObjectWidget> extends ProxyWidge
 abstract class InheritedWidget extends ProxyWidget {
   /// Abstract const constructor. This constructor enables subclasses to provide
   /// const constructors so that they can be used in const expressions.
-  const InheritedWidget({ Key key, Widget child })
+  const InheritedWidget({ Key key, this.scope, Widget child })
     : super(key: key, child: child);
 
+  /// An identifier that allows two differenciate between two [InheritedWidget]
+  /// of same type.
+  ///
+  /// If a [scope] is specified, that same object must be passed to
+  /// [Context.ancestorInheritedElementFroWidgetOfExactType] or
+  /// [Context.inheritFromWidgetOfExactType] to be able to obtain this
+  /// [InheritedWidget].
+  ///
+  /// Defaults to `null`.
+  final Object scope;
```

Comment:
Wondering if scope is the right word for this, it feels more like a key, maybe?

Reply:
"Key" is already taken. I would go for "context", but that's confusing too.

Reply:
That sounds more like some sort of a unique identifier/key, rather than scope, If what you're trying to solve with this PR is providing multiple widgets with same `runtimeType` then shouldn't the existing `key` be used for that? (i.e.: allowing users to uniquely identify a widget, either by specifying a `Key` or dynamically creating one if needed). This would end up being used like `Provider.of<SomeWidget>(context, someKey)`  maybe ?

Reply:
Not exactly. Keys have a vastly different use-case from scope. Two Theme may have be in the same scope, but have different keys.